homeassistant/components/media_player/ps4.py

```diff
@@ -0,0 +1,510 @@
+                        cover_art = game[cover]
+                        return cover_art
+
+    @property
+    def entity_picture(self):
+        """Return picture."""
+        if self._state == STATE_OFF:
+            return None
+
+        image_hash = self.media_image_hash
+        if image_hash is not None:
+            return ENTITY_IMAGE_URL.format(
+                self.entity_id, self.access_token, image_hash)
+
+        if self._media_content_id is None:
+            return None
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        return self._state
+
+    @property
+    def icon(self):
+        """Icon."""
+        return ICON
+
+    @property
+    def media_content_id(self):
+        """Content ID of current playing media."""
+        return self._media_content_id
+
+    @property
+    def media_content_type(self):
+        """Content type of current playing media."""
+        return MEDIA_TYPE_MUSIC
+
+    @property
+    def media_image_url(self):
+        """Image url of current playing media."""
+        if self._media_content_id is None:
+            return MEDIA_IMAGE_DEFAULT
+        try:
+            return self._media_image
+            # return self._gamesmap[self._media_content_id]
+        except KeyError:
+            return MEDIA_IMAGE_DEFAULT
+
+    @property
+    def media_title(self):
+        """Title of current playing media."""
+        return self._media_title
+
+    @property
+    def supported_features(self):
+        """Media player features that are supported."""
+        return SUPPORT_PS4
+
+    @property
+    def source(self):
+        """Return the current input source."""
+        return self._source
+
+    @property
+    def source_list(self):
+        """List of available input sources."""
+        return self._source_list
+
+    def turn_off(self):
+        """Turn off media player."""
+        self._ps4.standby()
+
+    def turn_on(self):
+        """Turn on the media player."""
+        self._ps4.wakeup()
+
+    def media_pause(self):
+        """Send keypress ps to return to menu."""
+        self._ps4.remote_control('ps')
+
+    def media_stop(self):
+        """Send keypress ps to return to menu."""
+        self._ps4.remote_control('ps')
+
+    def select_source(self, source):
+        """Select input source."""
+        if self._source_selected is not None:
+            _LOGGER.debug(
+                "Application %s is already in the process of starting (%s)",
+                self._source_selected, source)
+            return
+
+        for title_id, game in self._games.items():
+            if source == game:
+                _LOGGER.debug(
+                    "Starting PS4 game %s (%s) using source %s",
+                    game, title_id, source)
+                self._ps4.start_title(title_id)
+                return
+
+
+class Dummy():
```

Comment:
What's this used for?

If it's needed it has to be moved to a standalone library published on pypi. We don't want any device specific I/O code in home assistant.

https://developers.home-assistant.io/docs/en/creating_platform_code_review.html#6-communication-with-devices-services

Reply:
See docstring right below line 399.
"""The PS4 Credentials object."""

Reply:
> https://developers.home-assistant.io/docs/en/creating_platform_code_review.html#6-communication-with-devices-services
> '#bad
> status = requests.get(url('/status'))'

Class does not call an API. Rather it  simply uses UDP sockets.

Reply:
Transferred to pypi package.